straitlets/ext/tests/test_click.py

```diff
@@ -0,0 +1,285 @@
+from base64 import b64encode
+import re
+from textwrap import dedent
+
+import click
+from click.testing import CliRunner
+import pytest
+
+from straitlets import (
+    Serializable,
+    StrictSerializable,
+    Bool,
+    Unicode,
+    Integer,
+)
+from straitlets.ext.click import (
+    Base64ConfigFile,
+    JsonConfigFile,
+    YamlConfigFile,
+)
+from straitlets.test_utils import assert_serializables_equal
+
+
+@pytest.fixture
+def runner():
+    return CliRunner()
+
+
+class Config(Serializable):
+    bool = Bool()
+    unicode = Unicode()
+    int = Integer()
+
+
+class MissingAttr(Serializable):
+    bool = Bool()
+    unicode = Unicode()
+
+
+class StrictConfig(Config, StrictSerializable):
+    pass
+
+
+@pytest.fixture
+def expected_instance():
+    return Config(
+        bool=True,
+        unicode='ayy',
+        int=1,
+    )
+
+
+@pytest.fixture
+def missing_attr_instance():
+    return MissingAttr(
+        bool=True,
+        unicode='ayy',
+    )
+
+
+multi_error_output = re.compile(
+    dedent(
+        """\
+        ^Usage: main \[OPTIONS\]
+
+        Error: Invalid value for "--config": Failed to validate the schema:
+
+        bool:
+          No default value found for bool trait of <.+?>
+        int:
+          No default value found for int trait of <.+?>
+        unicode:
+          No default value found for unicode trait of <.+?>
+        $""",
+    ),
+)
+
+single_error_output = re.compile(
+    dedent(
+        """\
+        ^Usage: main \[OPTIONS\]
+
+        Error: Invalid value for "--config": Failed to validate the schema:
+          No default value found for int trait of <.+?>
```

Comment:
Oh, though I guess it's already in the message, which makes me wonder if the multiple error formatting is redundant...